tests/test_app.py

```diff
@@ -80,6 +80,20 @@ def test_is_logged_in_from_json_request(app, client):
 
 
 def test_logout(app, client, csrf_token_for):
+    callback_called = []
+
+    def callback1():
+        nonlocal callback_called
+        callback_called.append(True)
+
+    def callback2():
+        nonlocal callback_called
+        callback_called.append(True)
```

Comment:
For consistency, instead of using this architecture, let's refactor and use `unittest.mock` as we use in other parts of the tests.